lib/commands/web.js

```diff
@@ -834,6 +838,48 @@ const extensions = {
     }
   },
 
+  /**
+   * Calibrates web to real coordinates translation.
+   * This API can only be called from Safari web context.
+   * It must load a custom page to the browser, and then restore
+   * the original one, so don't call it if you can potentially
+   * lose the current web app state.
+   * The outcome of this API is then used in nativeWebTap mode.
+   * The returned value could also be used to manually transform web coordinates
+   * to real devices ones in client scripts.
+   *
+   * @this {XCUITestDriver}
+   * @returns {Promise<{offset: import('../types').Delta}>}
+   */
+  async mobileCalibrateWebToRealCoordinatesTranslation() {
+    if (!this.isWebContext()) {
+      throw new errors.NotImplementedError('This API can only be called from a web context');
+    }
+
+    const currentUrl = await this.getUrl();
+    await this.setUrl(`http://127.0.0.1:${this.opts.wdaLocalPort || 8100}/calibrate`);
```

Comment:
Kind of minor, but could we extract this url's host:port in a method and refer to one method from some places?
https://github.com/appium/appium-xcuitest-driver/blob/ff552e3d4ce09311926ad3c6de0c49557eb23de0/lib/driver.js#L478

Reply:
moved the logic the the separate `getWdaLocalhostRoot` method